ps3.py

```diff
@@ -322,27 +322,68 @@ def play_game(word_list):
             hand_match = deal_hand(HAND_SIZE)
-        if replay_the_hand == 'no' or hands_number == hands_number_copy_for_cycle:
+        # Main game
+        if substitute_letter_switch and saved_old_score == 'no_old_score':
+            # Possibility to substitute letter
+            # Starts if it wasn't used before and this match isn't a replay
             display_hand(hand_match)
             substitute_answer = input("Would you like to substitute a letter? ")
             if substitute_answer == 'yes':
+                # Turning off substitute function for the player
+                substitute_letter_switch = False
                 # Substituting letter in the hand, if user input is 'yes'
                 letter_substitute = input("Which letter would you like to replace: ")
+                print()
+                # Saving old hand in case of replaying current match
+                # Hand_without_substitute isn't 0 anymore, so program will use not modified hand during replay
+                hand_without_substitute = hand_match
                 hand_match = substitute_hand(hand_match, letter_substitute)
-        score_for_the_game += play_hand(hand_match, word_list)
-        hands_number -= 1
+            else:
+                # Needed for accurate output if user don't want to substitute letter
+                print()
+        else:
+            # This line is needed for situation, if hand_without_substitute not equal to zero
+            # But in this match substitute_letter_switch = False
+            # For not to start replaying with hand from old match
+            hand_without_substitute = 0
+        # Playing the match
+        score_match = play_hand(hand_match, word_list)
+        if saved_old_score != 'no_old_score':
+            # This part starts when the match is replaying
+            # Taking biggest score from during replaying the hand
+            score_match = max(saved_old_score, score_match)
+            # Returning old value of saved_old_score
+            saved_old_score = 'no_old_score'
+        score_for_the_game += score_match
         print("--------")
+        if replay_the_hand_switch:
+            # Possibility to replay the hand
+            replay_the_hand = input("Would you like to replay the hand? ")
+            if replay_the_hand == 'yes':
+                # Turning off replaying function for user
+                replay_the_hand_switch = False
+                # Replaying the hand
+                if hand_without_substitute != 0:
+                    # This lines is needed for replaying the hand without substitute
+                    # If hand was modified during this match
+                    # Hand_without_substitute is needed for checking if hand was modified during current game
+                    hand_match = hand_without_substitute
+                score_for_the_game -= score_match
+                saved_old_score = score_match
```

Comment:
Please - do operations with `score_for_the_game` in one place only

Reply:
done